tests/Unit/Utilities/Test_Functional.cpp

```diff
@@ -3,21 +3,196 @@
+  struct TestFuncEval##STRUCTNAME {                                        \
+    template <typename T, typename DistT, typename UniformGen>             \
+    void operator()(UniformGen gen, UniformCustomDistribution<DistT> dist, \
+                    T /*meta*/) noexcept {                                 \
+      auto val = make_with_random_values<typename T::type>(                \
+          make_not_null(&gen), make_not_null(&dist));                      \
+      CHECK(STRUCTNAME<>{}(val) == FUNC(val)); /*NOLINT*/                  \
+    }                                                                      \
+  }
+
+#define MAKE_BINARY_TEST(STRUCTNAME, FUNC)                                   \
+  struct TestFuncEval##STRUCTNAME {                                          \
+    template <typename T1, typename T2, typename DistT1, typename DistT2,    \
+              typename UniformGen>                                           \
+    void operator()(UniformGen gen, UniformCustomDistribution<DistT1> dist1, \
+                    UniformCustomDistribution<DistT2> dist2, T1 /*meta*/,    \
+                    T2 /*meta*/) noexcept {                                  \
+      auto val1 = make_with_random_values<typename T1::type>(                \
+          make_not_null(&gen), make_not_null(&dist1));                       \
+      auto val2 = make_with_random_values<typename T2::type>(                \
+          make_not_null(&gen), make_not_null(&dist2));                       \
+      CHECK(STRUCTNAME<>{}(val1, val2) == FUNC(val1, val2)); /*NOLINT*/      \
+    }                                                                        \
+  }
+
+#define MAKE_BINARY_OP_TEST(STRUCTNAME, OP)                                  \
+  struct TestFuncEval##STRUCTNAME {                                          \
+    template <typename T1, typename T2, typename DistT1, typename DistT2,    \
+              typename UniformGen>                                           \
+    void operator()(UniformGen gen, UniformCustomDistribution<DistT1> dist1, \
+                    UniformCustomDistribution<DistT2> dist2, T1 /*meta*/,    \
+                    T2 /*meta*/) noexcept {                                  \
+      auto val1 = make_with_random_values<typename T1::type>(                \
+          make_not_null(&gen), make_not_null(&dist1));                       \
+      auto val2 = make_with_random_values<typename T2::type>(                \
+          make_not_null(&gen), make_not_null(&dist2));                       \
+      CHECK(STRUCTNAME<>{}(val1, val2) == val1 OP val2); /*NOLINT*/          \
+    }                                                                        \
+  }
+
+#define MAKE_BINARY_INPLACE_TEST(STRUCTNAME, OP, TESTOP)                       \
+  struct TestFuncEval##STRUCTNAME {                                            \
+    template <typename T1, typename T2, typename DistT1, typename DistT2,      \
```

Comment:
I would find a comment explaining why the overloads exist instead of just one helpful :)